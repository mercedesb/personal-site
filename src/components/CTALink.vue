<template>
  <div :class="'CTA CTA--' + color">
    <a :href="url" :target="external ? '_blank' : '_self'">
      <div class='CTA-icon'>
        <img :src="icon" />
      </div>
      <h4 class='CTA-title'>{{ title }}</h4>
    </a>
  </div>
</template>

<script>
export default {
  props: {
    title: String,
    icon: String,
    url: String,
    color: {
      type: String,
      default: 'brown'
    },
    external: Boolean
  }
}
</script>

<style lang="scss" scoped>
  .CTA {
    flex: 1 1 50%;

    a {
      display:flex;
      align-items: center;
      margin: $base-spacing;
    }

    &-icon {
      background-color: $blue;
      margin-right: $small-spacing;
      padding: $small-spacing;
      max-width: 120px;

      img {
        width: 100px;
      }
    }

    @each $type in $colors-array {
      &--#{nth($type, 1)} {

        .CTA-icon {
          background-color: #{darken(saturate(nth($type, 2), 10%), 12%)};
        }

        a {
          border: 1px solid #{darken(saturate(nth($type, 2), 10%), 12%)};
        }
      }
    }
  }
</style>