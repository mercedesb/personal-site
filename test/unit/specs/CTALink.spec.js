import CTALink from '@/components/CTALink.vue'
import { TestUtility } from './TestUtility'

describe('CTALink', () => {
  let component

  const initialProps = {
    title: 'cta title',
    icon: 'http://google.com',
    url: 'cta',
    color: 'gray',
    external: false
  }

  const shallow = propsData => TestUtility.shallow(CTALink, {
    propsData: {
      ...initialProps,
      ...propsData
    }
  })

  describe('Snapshots', () => {
    describe('with a color passed in', () => {
      it('matches snapshot', () => {
        component = shallow()
        expect(component.element).toMatchSnapshot()
      })
    })

    describe('without a color passed in', () => {
      it('matches snapshot', () => {
        component = shallow({color: undefined})
        expect(component.element).toMatchSnapshot()
      })
    })
  })

  describe('Computed', () => {
    describe('modifierClass', () => {
      it('returns default if not external', () => {
        component = shallow()
        expect(component.vm.modifierClass).toEqual('default')
      })

      it('returns external if external', () => {
        component = shallow({external: true})
        expect(component.vm.modifierClass).toEqual('external')
      })
    })
  })
})
