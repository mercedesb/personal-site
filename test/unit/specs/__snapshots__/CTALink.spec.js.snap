// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTALink Snapshots with a color passed in matches snapshot 1`] = `
<div
  class="CTA CTA--gray CTA--default"
>
  <smart-link-stub
    class="CTA-link"
    to="cta"
  >
    <div
      class="CTA-icon"
    >
      <img
        alt="cta title icon"
        src="http://google.com"
      />
    </div>
     
    <h6
      class="CTA-title"
    >
      cta title
    </h6>
  </smart-link-stub>
</div>
`;

exports[`CTALink Snapshots without a color passed in matches snapshot 1`] = `
<div
  class="CTA CTA--brown CTA--default"
>
  <smart-link-stub
    class="CTA-link"
    to="cta"
  >
    <div
      class="CTA-icon"
    >
      <img
        alt="cta title icon"
        src="http://google.com"
      />
    </div>
     
    <h6
      class="CTA-title"
    >
      cta title
    </h6>
  </smart-link-stub>
</div>
`;
